### PR TITLE
[ads-facebook] upgrade facebook audience network SDK

### LINKED ISSUE #5028: expo ads facebook throwing error: The SDK version in the ad request is no longer supported
I have integrated facebook ads using Expo v33. but it is throwing error on app load **"The SDK version in the ad request is no longer supported for new apps. Please upgrade to one of the latest versions of the SDK"**.  next time it is throwing error like: **"Ad was re-loaded too frequently"**.
 I have seen same type of issue on github and stack overflow. one of the solution was upgrading the audiance network sdk to 5+ fix this issue. but Expo v33 already using `audience-network-sdk:5.1.1`. i already spent 6-7 hour on this issue and din't get any solution. any suggestion or solution will be appreciated.

check Reproducible Demo : [https://snack.expo.io/H1vxnXGfS](https://snack.expo.io/H1vxnXGfS)

**Error response** :
 `error Z {
  "_dispatchInstances": FiberNode {
    "tag": 5,
    "key": null,
    "type": "ViewManagerAdapter_CTKBannerView",
  },
  "_dispatchListeners": [Function onError],
  "_targetInst": FiberNode {
    "tag": 5,
    "key": null,
    "type": "ViewManagerAdapter_CTKBannerView",
  },
  "bubbles": undefined,
  "cancelable": undefined,
  "currentTarget": 257,
  "defaultPrevented": undefined,
  "dispatchConfig": Object {
    "registrationName": "onAdError",
  },
  "eventPhase": undefined,
  "isDefaultPrevented": [Function K],
  "isPropagationStopped": [Function K],
  "isTrusted": undefined,
  "nativeEvent": Object {
    "errorCode": 1203,
    "errorMessage": "The SDK version in the ad request is no longer supported for new apps. Please upgrade to one of the latest versions of the SDK",
  },
  "target": undefined,
  "timeStamp": 1563729189862,
  "type": undefined,
}`

### LINKED ISSUE #10081: Facebook SDK upgrade
Please provide the following:
1. SDK Version: 38.0.8
2. Android

I am making react native expo app. I wanted to add FB ads to my app. But I get this error

**The SDK version in the ad request is no longer supported for new apps. Please upgrade to one of the latest versions of the SDK**

When I run the app again, I get this one

error SyntheticEvent {
  "_dispatchInstances": FiberNode {
    "tag": 5,
    "key": null,
    "type": "ViewManagerAdapter_CTKBannerView",
  },
  "_dispatchListeners": [Function onError],
  "_targetInst": FiberNode {
    "tag": 5,
    "key": null,
    "type": "ViewManagerAdapter_CTKBannerView",
  },
  "bubbles": undefined,
  "cancelable": undefined,
  "currentTarget": 1575,
  "defaultPrevented": undefined,
  "dispatchConfig": Object {
    "registrationName": "onAdError",
  },
  "eventPhase": undefined,
  "isDefaultPrevented": [Function functionThatReturnsFalse],
  "isPropagationStopped": [Function functionThatReturnsFalse],
  "isTrusted": undefined,
  "nativeEvent": Object {
    "errorCode": 1002,
    "errorMessage": "Ad was re-loaded too frequently",
  },
  "target": undefined,
  "timeStamp": 1599433160964,
  "type": undefined,
}

### PR DESCRIPTION
**Pull request by cruzach**
# Why

closes https://github.com/expo/expo/issues/10081 and closes https://github.com/expo/expo/issues/5028

# How

Upgraded the SDK on android to 5.11.0. v6 is still very recently released (earlier this month) so I don't see a huge reason to upgrade to that version yet

Upgraded the SDK on iOS to 5.9.0. (this was a relatively small update). There's an issue affecting Audience Network SDK 5.10.0 for iOS, which results in the app crashing, so keeping our version below that

bare-expo doesn't use facebookads, so did not re-pod install there

# Test Plan

Tested NCL on iOS and Android, ads are populated 🙌

### CODE REVIEW

File: ios/Exponent.xcodeproj/project.pbxproj [hunk 1]
```diff
@@ -1172,7 +1173,7 @@
 		FC756CCA24CB542800665AE6 /* EXScopedNotificationSerializer.h */ = {isa = PBXFileReference; lastKnownFileType = sourcecode.c.h; path = EXScopedNotificationSerializer.h; sourceTree = "<group>"; };
 		FCA11CA524C1215B00C1FC54 /* EXScopedNotificationCategoriesModule.m */ = {isa = PBXFileReference; fileEncoding = 4; lastKnownFileType = sourcecode.c.objc; path = EXScopedNotificationCategoriesModule.m; sourceTree = "<group>"; };
 		FCA11CA624C1215B00C1FC54 /* EXScopedNotificationCategoriesModule.h */ = {isa = PBXFileReference; fileEncoding = 4; lastKnownFileType = sourcecode.c.h; path = EXScopedNotificationCategoriesModule.h; sourceTree = "<group>"; };
-		FE89956BF2504EB0A76B89AD /* MutableValueSetterProxy.h */ = {isa = PBXFileReference; explicitFileType = undefined; fileEncoding = 4; includeInIndex = 0; lastKnownFileType = sourcecode.c.h; name = MutableValueSetterProxy.h; path = MutableValueSetterProxy.h; sourceTree = "<group>"; };
+		FE89956BF2504EB0A76B89AD /* MutableValueSetterProxy.h */ = {isa = PBXFileReference; explicitFileType = undefined; fileEncoding = 4; includeInIndex = 0; lastKnownFileType = sourcecode.c.h; path = MutableValueSetterProxy.h; sourceTree = "<group>"; };
```
Comment (sjchmiela) [inline ios/Exponent.xcodeproj/project.pbxproj:1176]: What version of Xcode are you using? These look like rather safe changes, but if we don't need to introduce them if they eg. come from an old Xcode, let's not do it! 👍

File: ios/Podfile.lock [hunk 2]
```diff
@@ -4,7 +4,7 @@ PODS:
     - Google-Mobile-Ads-SDK
   - ABI36_0_0EXAdsFacebook (7.0.0):
     - ABI36_0_0UMCore
-    - FBAudienceNetwork (= 5.7.1)
```
Comment (sjchmiela) [inline ios/Podfile.lock:7]: 💅 💯

File: packages/expo-ads-facebook/ios/EXAdsFacebook.podspec [hunk 3]
```diff
@@ -2,6 +2,13 @@ require 'json'
 
 package = JSON.parse(File.read(File.join(__dir__, '..', 'package.json')))
 
+fb_audience_network_version = '~> 5.9.0'
+using_custom_fb_audience_network_version = defined? $FBAudienceNetworkVersion
+if using_custom_fb_audience_network_version
+  fb_audience_network_version = $AnalyticsVersion
```
Comment (sjchmiela) [inline packages/expo-ads-facebook/ios/EXAdsFacebook.podspec:8]:
```suggestion
  fb_audience_network_version = $FBAudienceNetworkVersion
```

File: packages/expo-ads-facebook/ios/EXAdsFacebook.podspec [hunk 4]
```diff
@@ -2,6 +2,13 @@ require 'json'
...
+  Pod::UI.puts "expo-ads-facebook: Using user specified Analytics version '#{$fb_audience_network_version}'"
```
Comment (sjchmiela) [inline packages/expo-ads-facebook/ios/EXAdsFacebook.podspec:9]:
```suggestion
  Pod::UI.puts "expo-ads-facebook: Using user specified FBAudienceNetwork version '#{$fb_audience_network_version}'"
```

File: packages/expo-ads-facebook/android/build.gradle [hunk 5]
```diff
@@ -56,7 +56,12 @@ if (new File(rootProject.projectDir.parentFile, 'package.json').exists()) {
           'This package can only be used in such projects. Are you sure you\'ve installed the dependencies properly?')
 }
 
+repositories {
+    mavenCentral()
+    jcenter()
```
Comment (sjchmiela) [inline packages/expo-ads-facebook/android/build.gradle:61]:
```suggestion
  mavenCentral()
  jcenter()
```

**Review by sjchmiela — APPROVED** [review]
Looks great! 🥇

File: ios/Exponent.xcodeproj/project.pbxproj (hunk 1 above)
Comment (cruzach) [reply ios/Exponent.xcodeproj/project.pbxproj:1176]: Xcode 12.0, maybe the current project is built on 11?

Comment (sjchmiela) [reply ios/Exponent.xcodeproj/project.pbxproj:1176]: I'm not sure it matters on which Xcode the project is built, what I meant is that Xcode did not propose to modify this file when I opened it on my computer and I was checking if something may be off, but if your installation is up-to-date then this is not a problem! — ✅

Comment (cruzach) [reply ios/Exponent.xcodeproj/project.pbxproj]: ah okay I see what you mean now, well I reverted the changes anyways so as long as that passes I'll merge it without these pbxproj changes included 👍

**Comment by abudhia-oyo** [comment]
Hi, please update the changelog so this could be merged to master.

**Comment by cruzach** [comment]
@abudhia-oyo this will go out for the managed workflow in the next patch release (not able to give you an ETA on that yet) but we should be able to publish a new version of `expo-ads-facebook` with this change which will be available for the bare workflow

**Comment by cruzach** [comment]
@abudhia-oyo this is now available in the bare workflow in `expo-ads-facebook@next`

**Comment by abudhia-oyo** [comment]
Thank @cruzach 

However, I'm on the managed workflow. Can you circle back with the team and give idea on a ballpark ETA? Will it take a few days/weeks/months to be available for managed workflow?

**Comment by cruzach** [comment]
you can follow along for that patch release here- https://github.com/expo/expo/issues/10464

**Comment by dodo4114** [comment]
Hi @cruzach. I am trying to add facebook ads  into my app for the first time. But I failed, and Facebook Support Team said  Audience Network SDK v6.1 is needed to me. Could you plz update SDK used for expo?

p.s.
Actually it's my first time to suggest. If there is something else to do for suggestion, tell me plz.

**Comment by cruzach** [comment]
v6.1 is the most recent release, they still support older releases

If you're in the managed workflow, you'll have to wait for the patch release as I said above

**Comment by brentvatne** [comment]
get this -> https://github.com/expo/expo/issues/10464#issuecomment-703178030

**Pull request merged**